### DISCUSSION TITLE
When No Workflow, Explicit or Inherited, Exists (UI | UX)

### DISCUSSION THREAD

**Discussion by markadrake** [nathanwoulfe/Plumber-2#137] [Ideas]
# 1. Explicit Error Messages

If a user tries to initiate a workflow on a content node but no workflow is available, they are met with the following error message:

> "ERROR: An error occurred initiating the workflow on node `####`"

I would suggest a rewrite of this message. I'm sure there are many ways to rewrite this message to provide more context. Here is what I'd suggest:

> "ERROR: Your request failed because there are no workflows attached to this content node."

![image](https://user-images.githubusercontent.com/713355/196582113-62b9f17b-6457-4a87-a7ac-ef1436c0e3d6.png)

# 2. Detect This Issue Early

Does any use case exist in which it makes sense to override the "Save and Publish" button, if a workflow does not exist? 

If no workflow is available:

- The main button could be restored to "Save and Publish" 
- The "Configuration" tab should be active when engaging with the Workflow content app.
- Hide "Active Workflow" button, until a workflow is detected.

![image](https://user-images.githubusercontent.com/713355/196582715-c7832600-e190-45e1-b1e4-37f379fb6542.png)

![image](https://user-images.githubusercontent.com/713355/196582894-00a5e377-3308-4b3b-9c0a-3dc51b96968d.png)

**Comment by nathanwoulfe** [comment]
Hey @markadrake I think this is due to my assumption that the root node would always have a workflow configured, but that isn't always the case (assumption based on the sites I've built/used, where the homepage was the root node, not a child of the root node).

This may already be fixed in the next release (v11/Umbraco.Workflow), will confirm and potentially port back down through 8/9 and 10.

**Reply by markadrake** [reply]
I wholeheartedly agree. I've made that recommendation to my client and will start at the root node in future projects. Thanks so much for the quick reply and best wishes for the new HQ release!

**Comment by nathanwoulfe** [comment]
On double checking, this appears to be fixed in v11, but I can't find exactly where the change was made. Issue would have been related to incorrectly checking whether a node has workflow configuration (likely fixed by a change to return an empty permissions list rather than null, given the client side checks the length of the permissions objects).

So long as you're planning a v11 upgrade, this will be available soon. For now though, adding a workflow group to the root node will prevent the error.
